File: CP3 JAVA/src/CP3/Professor.java
package CP3;

public class Professor extends Pessoa{
	
	private String disciplinas[];
	
	public Professor(){
		
	}
	
	public Professor(String nome, String ra, String disciplinas[]){
		super(nome, ra);
		this.disciplinas = disciplinas;
	}

	public String[] getDisciplinas() {
		return disciplinas;
	}

	public void setDisciplinas(String[] disciplinas) {
		this.disciplinas = disciplinas;
	}
}

File: C#/CP1/CP1-ClassLibrary/Models/Professor.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

using CP1_ClassLibrary;


namespace CP1_ClassLibrary.Models
{
    public class Professor : Funcionario
    {
        private String[] disciplinas;

        public Professor (
            string[] disciplinas, 
            string tempoContrato, 
            string cargo, 
            string nome, 
            string cpf
        ) : base(nome, cpf, tempoContrato, cargo)
        {
            this.disciplinas = disciplinas;
        }

        public String[] GetDisciplinas()
        {
            return this.disciplinas;
        }


        public void SetDisciplinas(String[] disciplinas)
        {
            this.disciplinas = disciplinas;
        }

    }
}
